Alerts

wontoan opened the pull request:
- [ ] Basic alerts.
- [ ] Default not hidden
- [ ] Would like to add icons when available

wontoan committed:
Add login and signup forms

vishalarora committed:
Set up style guide page

vishalarora committed:
Update color palette

vishalarora committed:
Merge pull request #10 from r-webdev-projects/colors

Update color palette

vishalarora committed:
Install autoprefixer

wontoan committed:
Added 4 different alert messages to styleguide

-Add .alert class for common layout.
-Add .alert-type class for each different alert
-Add basic styling.

vishalarora committed:
Add icon font and stylesheets

wontoan committed:
Style colors

wontoan committed:
Merge remote-tracking branch 'origin/styleguide-icons' into alerts

wontoan committed:
Edit for readibility

-Change colours to common names
-Tidy code formatting

wontoan committed:
Moved stylesheet to 'components'/alerts

vishalarora commented on app/assets/stylesheets/components/_alerts.scss:
```diff
@@ -0,0 +1,34 @@
+  padding: 15px;
+}
+
+.alert-success {
```
@wontoan There is a slight issue with code-reuse here.  The way you've written it, if I wanted to display a success alert, I'd have to do something like this:  `<div class='alert alert-success'>`.  That's not a horrible thing but it would be more readable to have either `class='alert success'` or `class='alert-success'`.  

For the first, you can use Sass nesting.  This is one of the most powerful parts of using a pre-processor.  For example:

```
.alert {
  border-radius: 10px;
  border-radius: 10px;
  color: $black;
  margin-bottom: 10px;
  padding: 15px;

  &.success {
    border: 1px solid $green;
  }
}
```

If you want to do the latter, you can use Sass's `@extends` feature.  For example:

```
.alert {
  // Your styles here
}

.alert-success {
  @extend .alert;
  border: 1px solid $green;
}
```

Read more here:  http://sass-lang.com/documentation/file.SASS_REFERENCE.html#extend

If you don't intend for `alert` to be a stand-alone class, then consider using a placeholder and extending it for the variations.  More info here:  http://thesassway.com/intermediate/understanding-placeholder-selectors

vishalarora commented on app/views/static_pages/styles.html.erb:
```diff
@@ -0,0 +1,78 @@
+
+  <div class="alert alert-success">
+    <strong>Success!</strong> Write your success message here.
+    <a href="#" class="close-alert">&times</a>
```
Technically, HTML entities would be written ending in a semicolon.  So this should be `&times;`.  Some browsers let you get away with forgetting the semicolon, some don't.  It's best to include it for that reason.

wontoan committed:
Update Alerts for code-reuse + edit entities.

Call “alert success” instead of “alert alert-success”.
Fixed entity to &times; for browser compatibility.

vishalarora commented on app/assets/stylesheets/components/_alerts.scss:
```diff
@@ -0,0 +1,34 @@
+    border: 1px solid $red;
+  }
+}
+
```
@wontoan Much better! There's just a couple nit-picky things that you can address but won't really make too much difference.

First, you have `background: $white` in all four of the alert states.  Since it applies to all of them,  you can just move it outside and apply it once to the main `.alert` class.

Second, you're also repeating `border: 1px solid …`.  This is more stylistic than anything though does have some implications in CSS specificity.  More importantly, there is no base style.  So, if someone forgets to add one of the state classes, the alert won't be appropriately styled.  So, I recommend adding `border: 1px solid $default-color` to the base style and then just use `border-color: $color` to change the color in each of the variations.  It also makes the code slightly more readable.

wontoan committed:
Move common styles to base alert

wontoan committed:
Default border-color.

vishalarora commented:
I'm closing this PR and opening another that merges this into `styleguide` instead of `master`.
